src/app/src/main/kotlin/org/imperial/mrc/hint/logging/LogMetadata.kt

```diff
@@ -0,0 +1,47 @@
+package org.imperial.mrc.hint.logging
+
+import org.imperial.mrc.hint.NoCoverage
+import javax.servlet.http.HttpServletRequest
+
+@NoCoverage
+data class Client(
+        val agent: String? = null,
+        val geoIp: String? = null,
+        val sessionId: String? = null
+)
+@NoCoverage
+data class AppOrigin(
+        val name: String? = "hint",
+        val type: String? = "backend"
+)
+@NoCoverage
+data class Request(
+        val method: String,
+        val path: String,
+        val hostname: String,
+        val client: Client
+)
+{
+        constructor(request: HttpServletRequest) : this(
+                request.method,
+                request.servletPath,
+                request.serverName,
+                Client(request.getHeader("User-Agent"), request.remoteAddr, request.session.id)
+        )
+}
+@NoCoverage
+data class Response(val message: String? = null, val status: String? = null)
+
+@NoCoverage
+data class ErrorMessage(val error: String? = null, val details: String? = null, val traceId: String? = null)
+
+@NoCoverage
+data class LogMetadata(
```

Comment:
I have just added NoCoverage annotation to LogMetadata because code coverage now complains after verifying the data object rather than verifying each object properties.